atc/worker/client.go

```diff
@@ -36,6 +36,19 @@ type Client interface {
 		filePath string,
 	) (io.ReadCloser, error)
 
+	RunCheckStep(
+		ctx context.Context,
+		logger lager.Logger,
+		owner db.ContainerOwner,
+		containerSpec ContainerSpec,
+		workerSpec WorkerSpec,
+		strategy ContainerPlacementStrategy,
+		containerMetadata db.ContainerMetadata,
+		resourceTypes atc.VersionedResourceTypes,
+		timeout time.Duration,
+		checkable resource.Resource,
+	) (result []atc.Version, err error)
```

Comment:
Should this return a CheckResult similar to all the other step interface methods?

Reply:
good ppoint!

Reply:
https://github.com/concourse/concourse/pull/5239/commits/c14cbf8230275ec71fd3bc124d89d4b4a7bffcff